### PR TITLE
hepnames: populate 667__a with previous_names

### PR DESCRIPTION
**Pull request by ammirate**

### CODE REVIEW

File: inspire_dojson/hepnames/rules.py [hunk 1]
```diff
@@ -224,6 +225,8 @@ def name2marc(self, key, value):
 
     if 'name_variants' in value:
         self['400'] = [{'a': el} for el in value['name_variants']]
+    if 'previous_names' in value:
+        self['667'] = {'a': ';'.join(value['previous_names'])}
```
Comment (michamos) [inline inspire_dojson/hepnames/rules.py]: I wouldn't join them, I would do
```suggestion
        self['667'] = [{'a': "Previously known as {}".format(prev_name)} for prev_name in value['previous_names']]
```

Comment (ammirate) [reply inspire_dojson/hepnames/rules.py]: Well, if you give a look to the data we have, that format doesn't really suite well. Some example of `667__a` field are:
* `Sakurai Prize 1988 Dirac Medal 1998` http://inspirehep.net/record/1018932/export/xm
* `2009 APS Dissertation Award in Nuclear Physics` http://inspirehep.net/record/1046396/export/xm
* `Do not confuse with Acharya, Bobby Samir` http://inspirehep.net/record/1060601/export/xm

Comment (ammirate) [reply inspire_dojson/hepnames/rules.py]: BTW, the `;` join is there because it is specified in the US https://its.cern.ch/jira/browse/INSPIR-2156

Comment (ammirate) [reply inspire_dojson/hepnames/rules.py]: Final decision is that we parse the `667__a` field reading the content only if the field contains `Formerly`.

File: tests/test_hepnames.py [hunk 2]
```diff
@@ -1619,6 +1619,33 @@ def test_public_notes_from_667__a():
...
+    ]
+    # record/1281982
+
+    metadata = {
+        'name': {
+            'value': 'Tomoko Ariga',
+            'previous_names': [
+                'Formerly Tomoko Furukawa',
+                'Formerly Second previous name',
+                'Nobel prize - this is a public note'
```
Comment (michamos) [inline tests/test_hepnames.py]: No, this should be
```
'previous_names': ['Tomoko Furukawa', 'Second previous name']
```

**Pull request merged**